clinica/pipelines/t1_freesurfer_atlas/t1_freesurfer_atlas_utils.py

```diff
@@ -21,45 +25,84 @@ def compute_atlas(
     for path in Path(path_to_atlas).rglob(
         "*" + to_process_with_atlases[0] + "_6p0.gcs"
     ):
-        hemisphere = Path(path.stem).stem
-        atlas_name = Path(path.stem).suffix[1:].split("_")[0]
-        sub, ses = to_process_with_atlases[1].split("_")
-        subject_dir = (
-            caps_directory
-            + "/subjects/"
-            + sub
-            + "/"
-            + ses
-            + "/t1/freesurfer_cross_sectional"
-        )
+        if "long" not in to_process_with_atlases[1]:
+            hemisphere = Path(path.stem).stem
+            atlas_name = Path(path.stem).suffix[1:].split("_")[0]
+            sub, ses = to_process_with_atlases[1].split("_")
+            subject_dir = (
+                Path(caps_directory)
+                / "subjects"
+                / sub
+                / ses
+                / ("t1/freesurfer_cross_sectional")
+            )
+            # path_to_freesurfer_cross = subject_dir + "/" + to_process_with_atlases[1]
+            path_to_freesurfer_cross = Path(subject_dir) / Path(
+                to_process_with_atlases[1]
+            )
+            output_path_annot = (
+                path_to_freesurfer_cross
+                / "label"
+                / (hemisphere + "." + atlas_name + ".annot")
+            )
+            sphere_reg = (
+                path_to_freesurfer_cross / "surf" / (hemisphere + ".sphere.reg")
+            )
+            if not os.path.isfile(sphere_reg):
+                cprint(
+                    f"The {hemisphere}.sphere.reg file appears to be missing. The data for {to_process_with_atlases[1]} will not be processed with {to_process_with_atlases[0]}",
+                    lvl="warning",
+                )
+            output_path_stats = (
+                path_to_freesurfer_cross
+                / "stats"
+                / (hemisphere + "." + atlas_name + ".stats")
+            )
+            subjid = to_process_with_atlases[1]
+            fake_dir = subject_dir
+        else:
+            hemisphere = Path(path.stem).stem
+            atlas_name = Path(path.stem).suffix[1:].split("_")[0]
+            sub, ses, long = to_process_with_atlases[1].split("_")
+            fake_dir = mkdtemp()
+            os.makedirs(fake_dir, exist_ok=True)
+            subjid = sub + "_" + ses + ".long." + sub + "_" + long
+            subject_dir = (
+                Path(caps_directory)
+                / "subjects"
+                / sub
+                / ses
+                / "t1"
+                / long
+                / "freesurfer_longitudinal"
+            )
+            path_to_freesurfer_cross = subject_dir / (
+                sub + "_" + ses + ".long." + sub + "_" + long
+            )
+            try:
+                os.symlink(path_to_freesurfer_cross, os.path.join(fake_dir, subjid))
```

Comment:
Refactoring?

Reply:
I refactored a few things, let me know if you think there is more to do !